pkg/controller/queuejobresources/genericresource/genericresource.go

```diff
@@ -450,3 +451,59 @@ func getContainerResources(container v1.Container, replicas float64) *clustersta
 	req.GPU = req.GPU * int64(replicas)
 	return req
 }
+
+//return value is a string for flexibility of expression
+func (gr *GenericResources) GetGenericItemKindStatus(aw *arbv1.AppWrapperGenericResource, namespace string) (completed string) {
+	dd := gr.clients.Discovery()
+	apigroups, err := restmapper.GetAPIGroupResources(dd)
+	if err != nil {
+		klog.Errorf("Error getting API resources, err=%#v", err)
+	}
+	restmapper := restmapper.NewDiscoveryRESTMapper(apigroups)
+	_, gvk, err := unstructured.UnstructuredJSONScheme.Decode(aw.GenericTemplate.Raw, nil, nil)
+	if err != nil {
+		klog.Errorf("Decoding error, please check your CR! Aborting handling the resource creation, err:  `%v`", err)
+	}
+	//service and podgroup has no conditions return ALWAYS true
+	//TODO: make it configurable?
+	if gvk.GroupKind().Kind == "Service" || gvk.GroupKind().Kind == "PodGroup" {
+		return "ignore"
+	}
+
+	mapping, err := restmapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		klog.Errorf("mapping error from raw object: `%v`", err)
+	}
+	restconfig := gr.kubeClientConfig
+	restconfig.GroupVersion = &schema.GroupVersion{
+		Group:   mapping.GroupVersionKind.Group,
+		Version: mapping.GroupVersionKind.Version,
+	}
+	rsrc := mapping.Resource
+	dclient, err := dynamic.NewForConfig(restconfig)
+	if err != nil {
+		klog.Errorf("Error creating new dynamic client, err=%#v", err)
+	}
+
+	inEtcd, err := dclient.Resource(rsrc).Namespace(namespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Error listing object: ", err)
+	}
+	for _, job := range inEtcd.Items {
+		completionRequiredBlock := aw.CompletionStatus
+		if len(completionRequiredBlock) > 0 {
+			conditions := job.Object["status"].(map[string]interface{})["conditions"].([]interface{})
```

Comment:
Since this logic is dealing with any object (including custom resources), should probably check to ensure there is a `conditions` field and likely to occur but also a check for `status` field

Reply:
I have seen status field has other objects, for instance, status may have `conditions` array or `loadbalancer` object, does it make sense to stick with `.status.conditions.type` for now?

Reply:
I'm ok with not adding the logic but a new test case should be added to handle such and event.  The test case should create an object without conditions that tracks.  For instance, duplicate on of the test cases and add an addition generic item that is a new namespace that tracks (`completionstatus`). This is a failure test case so the controller should handle this problem gracefully.